core/src/main/java/feign/RequestTemplate.java

```diff
@@ -215,7 +215,8 @@ public RequestTemplate resolve(Map<String, ?> unencoded) {
     for (String field : headers.keySet()) {
       Collection<String> resolvedValues = new ArrayList<String>();
       for (String value : valuesOrEmpty(headers, field)) {
-        String resolved = urlDecode(expand(value, encoded));
+        String expandedValueWithEncodedPlus = expand(value, encoded).replaceAll("\\+", "__PLUS__");
```

Comment:
you mind reader, you :)

nit: I think "replace" doesn't do regex, so you'd not have to escape the "+" iirc. Otherwise, add a comment like " // special case + as that's commonly used in vnd media types." as that helps set hack-context.